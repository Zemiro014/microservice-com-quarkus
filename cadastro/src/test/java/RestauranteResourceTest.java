

import javax.ws.rs.core.Response.Status;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.jeronimo.ifood.cadastro.CadastroTestLifecycleManager;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarRestaurantes() {
        String resultado = RestAssured.given()
                .when().get("/restaurantes")
                .then()
                .statusCode(Status.OK.getStatusCode())
                .extract().asString();
       Approvals.verifyJson(resultado);
    }
}
