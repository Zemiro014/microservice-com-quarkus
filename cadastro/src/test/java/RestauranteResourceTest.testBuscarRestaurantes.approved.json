[
  {
    "id": 123,
    "nomeFantasia": "Novo Restaurante 1",
    "proprietario": "ID do keyclock 1"
  },
  {
    "id": 234,
    "nomeFantasia": "Novo Restaurante 2",
    "proprietario": "ID do keyclock 2"
  }
]